modular_zubbers/code/modules/security/security_glock/gun.dm

```diff
@@ -1,28 +1,124 @@
 	icon = 'modular_zubbers/icons/obj/guns/sec_pistol.dmi'
-	icon_state = "silver"
+	icon_state = "black"
 	w_class = WEIGHT_CLASS_NORMAL
-	accepted_magazine_type = /obj/item/ammo_box/magazine/m9mm
-	spawn_magazine_type = /obj/item/ammo_box/magazine/m9mm
+	throwforce = 19 //Nokia of guns, no locker breaking though
+	force = 10 //same as baton
+	accepted_magazine_type = /obj/item/ammo_box/magazine/m9mm/security
+	spawn_magazine_type = /obj/item/ammo_box/magazine/m9mm/security
 	fire_sound = 'modular_zubbers/sound/weapons/gun/lock/shot.ogg'
-	fire_delay = 3
+	fire_delay = 5
 	can_suppress = FALSE
-	projectile_damage_multiplier = 0.75
+	projectile_damage_multiplier = 1
+	actions_types = list(/datum/action/item_action/toggle_mageject)
+	var/magejecting = 0 //whether we are launching the mags or not
 
-/obj/item/gun/ballistic/automatic/pistol/sec_glock/security //This is what you give to Security Officers.
-	name = "\improper C-CK 9x25mm"
-	desc = "The Compact Criminal Killer, or C-CK9 for short, is a semi-automatic ballistic pistol meant for regulated station defense. These are normally issued with a special firing pin that only allows firing on code blue or higher."
-	icon = 'modular_zubbers/icons/obj/guns/sec_pistol.dmi'
-	icon_state = "black"
-	pin = /obj/item/firing_pin/alert_level/blue
-	spawn_magazine_type = /obj/item/ammo_box/magazine/m9mm/flathead
-	fire_delay = 4
-	can_suppress = FALSE
-	projectile_damage_multiplier = 0.65
+/datum/action/item_action/toggle_mageject
+	button_icon = 'icons/obj/weapons/guns/ammo.dmi'
+	button_icon_state = "9x19p"
+	name = "Toggle magazine ejection"
+
+
+/obj/item/gun/ballistic/automatic/pistol/sec_glock/ui_action_click(mob/user, actiontype)
+	if(!istype(actiontype, /datum/action/item_action/toggle_mageject))
+		return ..()
+	magejecting = !magejecting
+	if(!magejecting)
+		balloon_alert(user, "not launching magazines")
+	else
+		balloon_alert(user, "launching empty magazines")
+
+	playsound(user, 'sound/items/weapons/empty.ogg', 100, TRUE)
+	update_appearance()
+	update_item_action_buttons()
+
+/obj/item/gun/ballistic/automatic/pistol/sec_glock/examine(mob/user)
+	. = ..()
+	. += span_notice("Using it in hand when out of ammo to start a spin reload. This can only be done if you have a spare magazine in or on your belt.")
+	. += span_notice("Spin reloads can be done on the move and need you only to hold your gun for their 2 second duration. ")
+	. += span_notice("If toggled on, spin reloads throw empty magazines in the direction you're facing, dealing damage dependant on their type. ")
+
+/obj/item/gun/ballistic/automatic/pistol/sec_glock/attack_self(mob/living/user)
+	var/obj/item/storage/our_belt = user.get_item_by_slot(ITEM_SLOT_BELT)
+	var/obj/item/ammo_box/magazine/m9mm/security/swapped_mag = locate(/obj/item/ammo_box/magazine/m9mm/security) in our_belt.contents
```

Comment:
```suggestion
	var/obj/item/belt_item = user.get_item_by_slot(ITEM_SLOT_BELT)
	var/obj/item/ammo_box/magazine/swapped_mag = istype(belt_item, accepted_magazine_type) ? belt_item : locate(accepted_magazine_type) in belt_item
```
Previous code failed to account for the case where you're wearing a magazine directly in your belt slot